sqlx-core/src/error.rs

```diff
@@ -138,9 +138,8 @@ impl Error {
 pub(crate) fn mismatched_types<DB: Database, T: Type<DB>>(ty: &DB::TypeInfo) -> BoxDynError {
     // TODO: `#name` only produces `TINYINT` but perhaps we want to show `TINYINT(1)`
     format!(
-        "mismatched types; Rust type `{}` (as SQL type `{}`) is not compatible with SQL type `{}`",
```

Comment:
I'm loathe to lose this information as I think it could be really helpful with debugging.

I think instead we should fix this one level up, calling this function with the underlying `DB` type instead of `Any`:

* Override [`Row::try_get()`](https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/row.rs#L124) in [the impl for `AnyRow`](https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/any/row.rs#L40)
* Override [`Value::try_decode`](https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/value.rs#L71) in [the impl for `AnyValue`](https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/any/value.rs#L56)

I'd probably make a version of this function as a method on [`AnyTypeInfo`](https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/any/type_info.rs) and match on `AnyTypeInfoKind`.